### DISCUSSION TITLE
Allow to always create workspace above

### DISCUSSION THREAD

**Discussion by kchibisov** [YaLTeR/niri#79] [Ideas]
I don't see why not, I feel like the way they're only bottom is a bit weird.

The only issue is absolute indexing to get into workspace, but we can sort of shift forward until we reach the limit?

**Comment by YaLTeR** [comment]
I can add it and we can see how it goes. Will need some careful handling wrt. layout invariants in the code but I think I have plenty of test coverage there.

For absolute indexing, maybe we could bind `0` to it?

**Reply by kchibisov** [reply]
>For absolute indexing, maybe we could bind 0 to it?

Don't understand this part? The issue is how bindings with numbers should work to focus, I think they should just shift? But creating a workspace. Also, do you want a special binding to create workspace or just `Up` (similar how `Down`) focus and it dynamically being created?

**Reply by YaLTeR** [reply]
Did I understand right that the suggestion is to always have one empty workspace above, like how there's currently always one empty workspace below? In this case I think it would make sense to bind the `0` key to index it, so that `1` remains the first actual workspace with windows.

**Reply by kchibisov** [reply]
Yeah, should work fine.

**Comment by famiu** [comment]
Any updates on this?

**Reply by YaLTeR** [reply]
https://yalter.github.io/niri/Configuration%3A-Layout.html#empty-workspace-above-first
